### PR TITLE
Move TravisCI Build to GH Actions

### LINKED ISSUE #349: Move the Travis-CI build to GitHub Actions
### Information

Currently, we build all the connext_dds examples on Travis-CI. It will be nice to have everything on GitHub Actions and remove the Travis-CI dependency.

### LINKED ISSUE #397: Problems with GH Actions
### What is the current behavior?

Secrets cannot be shared in Pull Requests workflows.

### Steps to reproduce the issue

Take a look of #349 (you can see in the steps the secret is empty). Also, we tried to reproduce the issue in [another repo](https://github.com/ManuelJNunez/trying/pull/1).

### More information

> Secrets are environment variables that are encrypted. Anyone with collaborator access to this repository can use these secrets for Actions.

> Secrets are not passed to workflows that are triggered by a pull request from a fork.

In [this thread](https://github.community/t/make-secrets-available-to-builds-of-forks/16166) you can also get more information.

### PR DESCRIPTION
**Pull request by ManuelJNunez**
:white_check_mark: I have updated the documentation accordingly.
:white_check_mark: I have read the CONTRIBUTING document.

### Summary

I created a GH Actions workflow that builds all the examples and executes the static analysis. Also, I removed the execution of the linters from this workflow because there are others that execute them.

### Details and comments

close #349
close #397

### CODE REVIEW

**Comment by ManuelJNunez** [comment]
It's necessary to configure the secrets in this repository to make the workflow work properly.

**Comment by iblancasa** [comment]
@fgaranda and @lulivi can you help @ManuelJNunez with this?

**Review by fgaranda — APPROVED** [review]

**Comment by lulivi** [comment]
It seems to be working fine, right? no errors or warning at all. Pretty cool!

Should we fail the build if any of the required environment variables are not defined? Instead of Skipping it?

**Comment by ManuelJNunez** [comment]
Actually there is a problem, but it will be fixed soon. Take a look at #397

**Comment by ManuelJNunez** [comment]
The other workflow will not run until this PR is merged.

**Comment by ManuelJNunez** [comment]
close #397

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            })[0];
+            var download = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            fs.writeFileSync('${{github.workspace}}/pr.zip', Buffer.from(download.data));
+      - run: unzip pr.zip
+        if: ${{ github.event_name == 'workflow_run' }}
```
Comment (lulivi) [inline .github/workflows/build.yml:50]:
```suggestion
      - name: Unzip artifact
        if: ${{ github.event_name == 'workflow_run' }}
        run: unzip pr.zip
```
To follow a convention

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            await github.issues.createComment({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: issue_number,
+              body: `The compilation is starting. Take a look
+                    [here](https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}).`
+            });
+      - uses: actions/checkout@v2
+        if: ${{ github.event_name == 'push' }}
+      - uses: actions/checkout@v2
```
Comment (lulivi) [inline .github/workflows/build.yml:71]: One of these two `- uses` is not correct. Maybe the first one?

**Review by lulivi — CHANGES_REQUESTED** [review]

**Review by lulivi — APPROVED** [review]

**Comment by lulivi** [comment]
I'm not sure why the build is not triggering... maybe we need to merge it in order for the build to trigger? The thing is that the artifact uploading is trigerring.

**Comment by ManuelJNunez** [comment]
@lulivi You're right, the other workflow is executed in the context of this repository (cannot be executed on forks), so it has to be merged.

**Comment by lulivi** [comment]
But the purpose is for it to execute on pull requests, too, right?

**Comment by ManuelJNunez** [comment]
@lulivi It will be executed in the next Pull Request if this one is merged, it cannot be executed in this fork.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
+name: build
```
Comment (angelrti) [inline .github/workflows/build.yml:1]: this is a test message

**Review by angelrti — CHANGES_REQUESTED** [review]

**Pull request merged**